main/middleware/apisix_user.py

```diff
@@ -33,7 +49,77 @@ def decode_apisix_headers(request, header):
         return None
 
     log.debug("decode_apisix_headers: Got %s", apisix_result)
-    return {"username": apisix_result.get("preferred_username", None)}
+    return {
+        modelKey: apisix_result[data_mapping[modelKey]]
+        for modelKey in data_mapping
+        if data_mapping[modelKey] in apisix_result
+    }
+
+
+def get_user_from_apisix_headers(request, decoded_headers, original_header):
+    """
+    Get a user based on the APISIX headers, create user/profile if needed.
+
+    Args:
+        request: Django request object
+        decoded_headers: Decoded APISIX headers
+        original_header: Original header
+
+    Returns:
+        User object
+
+    """
+
+    User = get_user_model()
+
+    if not decoded_headers:
+        return None
+
+    global_id = decoded_headers.get("global_id", None)
```

Comment:
If the global id is `None`, we should log an error and return as if there were no headers, otherwise we'll end up creating a user per request.